ruby-server-example/Dockerfile

```diff
@@ -0,0 +1,22 @@
+# Ruby Alpine image (after using bsf dockerfile digest)
+FROM ruby:3.1-alpine AS build
+
+# Install all required dependencies for Ruby gems
+RUN apk add --no-cache build-base libxml2-dev libxslt-dev tzdata
+
+WORKDIR /src
+
+# Copy the Gemfile and Gemfile.lock into the container
+COPY Gemfile Gemfile.lock /src/
+
+# Install the required gems
+RUN bundle install
```

Comment:
Can we vendor Ruby depenencies and install them via it?
https://stackoverflow.com/questions/21547390/how-to-put-ruby-dependencies-in-local-project-folder

Reply:
Yeah we can do this, but for the current example we just require this Dockerfile (I just verified)
```Dockerfile
FROM ttl.sh/ruby-server:dev AS build

# Expose the application port
EXPOSE 9898

# Command to run the Ruby application
CMD ["ruby", "main.rb"]
```